core/src/main/scala/cats/syntax/alternative.scala

```diff
@@ -13,11 +13,20 @@ trait AlternativeSyntax {
     new GuardOps(b)
 }
 
-final class UniteOps[F[_], G[_], A](private val fga: F[G[A]]) extends AnyVal {
```

Comment:
**Note**: the `fga`'s access is changed from `private` to `protected` because `fga` needs to be accessible from `UniteOpsBinCompat0`. The same change is made for the `SeparateOps` below.

**One more note**: the `UniteOps`'s constructor is made `private[syntax]`. That makes sense in general – I believe it is better to keep its access scope as narrow as possible. And apparently it does not affect the binary compatibility. But if someone has a concern regarding this change – speak up please, we can change this access back to `public`.

Reply:
The `fga` seems fine.

Changing the constructor visibility is a source-breaking change that's highly unlikely to break anybody's source.  I'd probably leave it public to be conservative in Cats, but I'm not too fussed either way.